### PR TITLE
add fix to not LSP documentation with tsserver lsp

### PR DESCRIPTION
**Pull request by greeschenko**
this change prevent lazy load documentation request to LSP server for items that already  consist documentation 

fix issue https://github.com/yegappan/lsp/issues/215

### CODE REVIEW

File: autoload/lsp/completion.vim [hunk 1]
```diff
@@ -248,6 +248,74 @@ export def CompletionReply(lspserver: dict<any>, cItems: any)
   endif
 enddef
 
+# process the completion documentation without LSP server request
+def ShowCompletionDocumentationNoLsp(cItem: any)
```
Comment (yegappan) [inline autoload/lsp/completion.vim:252]: It looks like most of the code in the CompletionResolveReply() function is duplicated here.  Can you refactor the CompletionResolveReply() function to use this function?

**Review by yegappan — CHANGES_REQUESTED** [review]

File: autoload/lsp/completion.vim [hunk 2]
```diff
@@ -269,13 +268,6 @@ export def CompletionResolveReply(lspserver: dict<any>, cItem: any)
   var infoText: list<string>
   var infoKind: string
 
-  if cItem->has_key('detail') && !cItem.detail->empty()
```
Comment (yegappan) [inline autoload/lsp/completion.vim:272]: Why is this block of code removed?

**Review by yegappan — CHANGES_REQUESTED** [review]

File: autoload/lsp/completion.vim [hunk 3]
```diff
@@ -248,9 +248,8 @@ export def CompletionReply(lspserver: dict<any>, cItems: any)
   endif
 enddef
 
-# process the 'completionItem/resolve' reply from the LSP server
-# Result: CompletionItem
-export def CompletionResolveReply(lspserver: dict<any>, cItem: any)
+# process the completion documentation without LSP server request
```
[the remark this answers is not in the document]
Comment (greeschenko) [reply autoload/lsp/completion.vim]: add fix in https://github.com/yegappan/lsp/pull/230/commits/1888f32088a5303c07d009b283f58b6b6a1e130b

File: autoload/lsp/completion.vim (hunk 2 above)
Comment (greeschenko) [reply autoload/lsp/completion.vim]: sorry my mistake  
fixed in https://github.com/yegappan/lsp/pull/230/commits/1888f32088a5303c07d009b283f58b6b6a1e130b

**Pull request merged**